### PR TITLE
UTF-8 validation

### PR DESCRIPTION
**Pull request by sdkrystian**

### CODE REVIEW

File: include/boost/json/basic_parser.hpp [hunk 1]
```diff
@@ -52,6 +53,12 @@ enum class basic_parser::state : char
     str5, str6, str7,
     sur1, sur2, sur3, sur4,
     sur5, sur6,
+    utf1, utf2, utf3, 
+    utf4, utf5, utf6,
+    utf7, utf8, utf9, 
+    utf10, utf11, utf12, 
+    utf13, utf14, utf15, 
+    utf16, utf17, utf18,
```
Comment (vinniefalco) [inline include/boost/json/basic_parser.hpp:61]: column alignment TRIGGERS ME!!

Comment (sdkrystian) [reply include/boost/json/basic_parser.hpp:61]: Hah, I tried

Comment (vinniefalco) [reply include/boost/json/basic_parser.hpp:61]:
```
    utf1,  utf2,   utf3, 
    utf4,  utf5,   utf6,
    utf7,  utf8,   utf9, 
    utf10, utf11, utf12, 
    utf13, utf14, utf15, 
    utf16, utf17, utf18,
```

File: include/boost/json/detail/basic_parser.hpp [hunk 2]
```diff
@@ -121,26 +121,30 @@ class basic_parser
...
     
+    template<bool StackEmpty>
+    result validate_utf8(const_stream& cs);
+
     template<bool StackEmpty, class Handler>
     result parse_document(Handler& h, const_stream& cs);
     
     template<bool StackEmpty, bool AllowComments,
-        bool AllowTrailing, class Handler>
+        bool AllowTrailing, bool NoValidate, class Handler>
```
Comment (vinniefalco) [inline include/boost/json/detail/basic_parser.hpp:135]: `AllowInvalid` - see the pattern? :)

File: include/boost/json/parse_options.hpp [hunk 3]
```diff
@@ -23,23 +23,30 @@
...
 {
 #if __cplusplus < 201402L
     parse_options(
         bool comments = false, 
-        bool commas = false) noexcept
+        bool commas = false,
+        bool no_validate = false) noexcept
         : allow_comments(comments)
         , allow_trailing_commas(commas)
+        , skip_utf8_validation(no_validate)
```
Comment (vinniefalco) [inline include/boost/json/parse_options.hpp:39]: Something like
```
    parse_options(
        bool allow_comments_ = false, 	        bool comments = false, 
        bool allow_trailing_ = false) noexcept	        bool commas = false,
        bool allow_invalid_utf8_ = false) noexcept
        : allow_comments(allow_comments_)
        , allow_trailing(allow_trailing_)
        , allow_invalid_utf8(allow_invalid_utf8_)
```

**Pull request merged**